xmtp_mls/src/groups/mod.rs

```diff
@@ -364,15 +364,9 @@ fn build_protected_metadata_extension(
         identity.account_address.clone(),
         policies,
     );
-    let protected_metadata = ProtectedMetadata::new(
-        &identity.installation_keys,
-        identity.application_id(),
-        identity.credential()?,
-        identity.installation_keys.to_public_vec(),
-        metadata.try_into()?,
-    )?;
-
-    Ok(Extension::ProtectedMetadata(protected_metadata))
+    let protected_metadata = Metadata::new(metadata.try_into()?);
```

Comment:
nit about naming:   If its no longer instantiating a `ProtectedMetadata`, as that's being replaced with an `ImmutableMetadata`.

```suggestion
    let immutable_metadata = Metadata::new(metadata.try_into()?);
```